### ISSUE TITLE
No mention of Security

### ISSUE THREAD

**Issue by frio** [thedaviddias/Front-End-Checklist#23]
I can't help but notice the lack of a section on Security.  As a suggestion, it may be worth adding some mention of SRI attributes where applicable, and ensuring CSP directives (alongside other useful stuff like CSRF protections) are included.

**Comment by collinwu** [comment]
Was going to mention there's room for a security section

**Comment by thedaviddias** [comment]
@frio @collinwu indeed, that part is missing. Let me add that section and don't hesitate to comment on that.

**Comment by thedaviddias** [comment]
I just add a first draft for the security section, don't hesitate to give me a feedback or make some changes.

**Comment by frio** [comment]
Brilliant!  For reference, my little `nonfunctional.md` file contains the following points for security:

 * HSTS ✅ 
 * CSRF ✅ 
 * CORS ✅ 
 * XSS protection ✅ 
 * Content Security Policy
 * SRI hashes where possible

You've got stuff I'm missing (like X-Frame-Options), so I'll quietly copy/paste those in :).  SRI and Content Security Policy are both probably worth a mention :) -- CSP is probably a "high" and SRI is probably a "low".

Possibly one of the cheat sheets for XSS and similar mentions CSP, but it's worth a top-level mention IMHO.

I should've said in the first comment, but thanks for building this handy little resource.  The list of so-called `non-functional` requirements for a modern webapp is vast (and seemingly grows forever), so it's great to have a good enumeration of them.

**Comment by thedaviddias** [comment]
Don't hesitate to propose (in a PR) a change in the priority if you don't agree. Thanks for your support @frio (frio means cold in portuguese :D). Hopefully, that checklist will improve a lot in the next days.

**Comment by praveeno** [comment]
+1

**Comment by thedaviddias** [comment]
@collinwu @frio Some time ago, we added a security section, but since I was exchanging with some developers, and I would have your POV. A Front-End developer is in general not responsible for server-side security rules... If you develop an application with server-side, you may be more a Full-Stack developer than a Front-End Dev... Don't you think?

**Comment by thedaviddias** [comment]
Ping @collinwu @frio

**Comment by frio** [comment]
Phew, sorry for the slow reply!  Security is a nightmare in our industry; we find new flaws with potentially dangerous implications day in and day out.  There are heaps of things that front-end developers should be aware of in the security space :)!

At a minimum, I'd expect a frontend developer to at least be aware of the [OWASP Top 10](https://www.owasp.org/index.php/OWASP_Top_Ten_Cheat_Sheet).  You're right that they may not need to know how to solve all of those, but I'd expect them to know how to protect against XSS, for instance, or CSRF.  There are also nice little tweaks (like SRI hashes) that help to prevent against a compromised CDN or ad from subverting your site.

Basically -- yes, full-stack devs need to know more :).  However, I'd argue that front-end devs need to understand the world they deploy into at a minimum, and understand how to address some of the flaws that most impact them.
